src/boot/bootctl-random-seed.c

```diff
@@ -17,6 +18,39 @@
 #include "tmpfile-util.h"
 #include "umask-util.h"
 
+static int random_seed_verify_permissions(int fd, mode_t expected_type) {
+        _cleanup_free_ char *full_path = NULL;
+        struct stat st;
+        int r;
+
+        assert(fd >= 0);
+
+        r = fd_get_path(fd, &full_path);
+        if (r < 0)
+                return log_error_errno(r, "Unable to determine full path of random seed fd: %m");
+
+        if (fstat(fd, &st) < 0)
+                return log_error_errno(errno, "Unable to stat %s: %m", full_path);
+
+        if (((st.st_mode ^ expected_type) & S_IFMT) != 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EBADF),
```

Comment:
Shouldn't this be EBADFD instead?

Reply:
EBADFD is a bit weird, because the string for it is "File descriptor in bad state", but that's not really the case here, it's simply of the wrong kind, not in a bad state..

EBADF is more generic, it just means "Bad file descriptor". And that certainly applies here.

other syscalls in similar siutations sometimes return EBADF and sometimes EINVAL...

for example inotify_add_watch() returns EBADF if pass some non-inotify fd as first argu. otoh readlinkat() returns EINVAL if the fd doesn't refer to a symlink.

Of the two I think EINVAL is worse though as it is even less accurate... SO i opted for EBADF